Plug-in fails with cryptic error on non-JDK 1.8 machines.

Running the Infer plug-in on a machine with JDK 7 - I get this error:
```
Caused by: com.google.common.util.concurrent.ExecutionError: java.lang.UnsupportedClassVersionError: com/uber/infer/InferAndroidPlugin : Unsupported major.minor version 52.0
```

I think we need to make sure we explicitly support pre-JDK 1.8 - should be an easy fix.

--- a/infer-plugin/build.gradle
+++ b/infer-plugin/build.gradle
@@ -1,5 +1,8 @@
 apply plugin: 'groovy'
 
+sourceCompatibility = JavaVersion.VERSION_1_7
+targetCompatibility = JavaVersion.VERSION_1_7
+
 repositories {
     jcenter()
 }

